File: python/hashTable/hashTable.py

class HashTable:

    def __init__(self):
        self.table = {}
    
    def add(self, key, value):
        self.table[key] = value

    def exists(self, key):
        returnValue = False
        if self.table.get(key) is not None:
            returnValue = True
        return returnValue

    def get(self, key):
        returnValue = None
        if self.table.get(key) is not None:
            returnValue = self.table[key]
        return returnValue

    def remove(self, key):
        return self.table.pop(key)

    def returnTable(self):
        return self.table

    def removeAllitems(self):
        self.table.clear()
        self.returnTable()

File: cpp/hashTable/hashTable.cpp
#include <string.h>
#include <iostream>
#include "hashTable.h"


// https://www.digitalocean.com/community/tutorials/hash-table-in-c-plus-plus

hashTable::hashTable(int c)
{
    this->CAPACITY = c;
}

void hashTable::add(int key, int value)
{
    std::size_t str_hash = std::hash<int>{} (key);
}

void hashTable::add(std::string key, int value)
{
    std::size_t str_hash = std::hash<std::string>{} (key);
}
int hashTable::exists(int key)
{
}
void *hashTable::get(int key)
{
}
void *hashTable::remove(int key)
{

}